cmd/collector/app/server/grpc.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/grpclog"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/handler"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
+)
+
+// GRPCServerParams to construct a new Jaeger Collector gRPC Server
+type GRPCServerParams struct {
+	TLSConfig     tlscfg.Options
+	Port          int
+	Handler       *handler.GRPCHandler
+	SamplingStore strategystore.StrategyStore
+	Logger        *zap.Logger
+	OnError       func(error)
+}
+
+// StartGRPCServer based on the given parameters
+func StartGRPCServer(params *GRPCServerParams) (*grpc.Server, error) {
+	var server *grpc.Server
+	grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, os.Stderr, os.Stderr))
```

Comment:
hm, odd that we don't do that in query-service, as far as I could find

Reply:
I think this might actually be more suitable for the mains...